src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -258,8 +257,15 @@ public function getBoolean($key, $default = false, $deep = false)
      *
      * @return mixed
      */
-    public function filter($key, $default = null, $deep = false, $filter = FILTER_DEFAULT, $options = array())
+    public function filter($key, $default = null, $filter = FILTER_DEFAULT, $options = array())
     {
+        $deep = false;
+        if (is_bool($filter)) {
```

Comment:
Please add `@trigger_error('The "deep" argument of '.__METHOD__.' is deprecated since version 2.8 and will be removed in 3.0.', E_USER_DEPRECATED);` in this condition.